unity/src/main/java/io/delta/unity/UCCatalogManagedCommitter.java

```diff
@@ -273,4 +359,20 @@ private static CommitFailedException storageCFEtoKernelCFE(
         storageCFE.getMessage(),
         storageCFE.getCause());
   }
+
+  /**
+   * Returns true if the provided Throwable is to be considered fatal, or false if it is to be
+   * considered non-fatal
+   */
+  private static boolean isFatal(Throwable t) {
+    // VirtualMachineError includes OutOfMemoryError and other fatal errors
+    if (t instanceof VirtualMachineError
+        || t instanceof ThreadDeath
+        || t instanceof InterruptedException
```

Comment:
why is interruptedexception considered fatal?

Reply:
cc @xzhseh can you help respond?